src/coreclr/jit/lowerarmarch.cpp

```diff
@@ -464,6 +456,17 @@ void Lowering::LowerBlockStore(GenTreeBlk* blkNode)
         {
             blkNode->gtBlkOpKind = GenTreeBlk::BlkOpKindUnroll;
 
+            if (!isSrcAddrLocal || !isDstAddrLocal)
```

Comment:
All these changes are specific to arm64. Shouldn't it be ifdef'ed so it doesn't affect arm32?

Reply:
hhm, today `blkNode->gtBlkOpGcUnsafe = true` is marked for both arm/arm64. This PR just extends it to more scenarios?